packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/client_config.ts

```diff
@@ -23,8 +23,8 @@ import { DEFAULT_HEADERS } from './headers';
 export function parseClientOptions(
   config: ElasticsearchClientConfig,
   scoped: boolean
-): ClientOptions {
-  const clientOptions: ClientOptions = {
+): ClientOptions & { agent: HttpAgentOptions } {
+  const clientOptions: ClientOptions & { agent: HttpAgentOptions } = {
```

Comment:
NIT: I would introduce an explicit type instead of an inline def

```ts
export type ParsedClientOptions = Omit<ClientOptions, 'agent'> & { agent: HttpAgentOptions }
```